cloudwatch_monitoring/state_machine.py

```diff
@@ -0,0 +1,140 @@
+            state_machine_widgets.append(state_machine_widget)
+            positioning.iterate_positioning()
+
+    return state_machine_widgets
+
+
+def get_all_state_machines(region):
+    """
+    Using the AWS python sdk (boto3), grab all the state machines for the specified account for a given region.
+
+    :param region: The region, for us that's usually us-west-2
+    :return: response: a page of state machines in the account.
+    :rtype: dict
+    """
+    sfn_client = boto3.client("stepfunctions", region_name=region)
+
+    # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/stepfunctions.html#SFN.Client.list_state_machines
+    # TODO this pagination logic exists in other modules as well, consider moving it into its own utility
+    # TODO module or trying to get a proper boto3 paginator to work...
+    response = {}
+    next_token = None
+    while True:
+        if next_token:
+            response_iterator = sfn_client.list_state_machines(
+                    # maxResults has to be set in order to receive a pagination token in the response
+                    maxResults=10,
+                    nextToken=next_token)
+            response['stateMachines'].extend(response_iterator['stateMachines'])
+        else:
+            response_iterator = sfn_client.list_state_machines(
+                    maxResults=10
+            )
+            response.update(response_iterator)
+        try:
+            next_token = response_iterator['nextToken']
+        except KeyError:
+            # no more pages, move on
+            break
+
+    return response
+
+
+def is_iow_state_machine_filter(state_machine_arn, deploy_stage, region):
+    """
+    Apply filters to determine if the state machine is a tagged IOW asset in the correct tier.
+
+    :param state_machine_arn: A single state machine arn
+    :param deploy_stage: The specified deployment environment (DEV, TEST, QA, PROD-EXTERNAL)
+    :param region: typically 'us-west-2'
+    :return: is_iow_state_machine: is this an IOW state machine or not
+    :rtype: bool
+    """
+    sfn_client = boto3.client("stepfunctions", region_name=region)
+
+    is_iow_state_machine = False
+
+    # filtering on deploy tier, which we capitalize
+    if deploy_stage.upper() in state_machine_arn:
+        # launch API call to grab the tags for the state machine
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/stepfunctions.html#SFN.Client.list_tags_for_resource
+        state_machine_tags = sfn_client.list_tags_for_resource(resourceArn=state_machine_arn)
+
+        # we only want state machines that are tagged as 'IOW'
+        if 'tags' in state_machine_tags:
+            for tag in state_machine_tags['tags']:
+                if 'key' in tag:
+                    if 'wma:organization' in tag['key']:
+                        if 'IOW' == tag['value']:
+                            is_iow_state_machine = True
+
+    return is_iow_state_machine
```

Comment:
Consider putting both of these functions in a class. That way the can share attributes and variables (i.e the boto3 client, region name).

Reply:
Thanks @ayan-usgs - going to merge this one and address as a separate PR - it's a pattern that I can add to the other asset modules as too.